src/content/developers/tutorials/scam-token-tricks/index.md

```diff
@@ -0,0 +1,471 @@
+        _f_(sender, recipient, amount);
+        _approve(sender, _msgSender(), _allowances[sender][_msgSender()].sub(amount, "ERC20: transfer amount exceeds allowance"));
+        return true;
+    }
+```
+
+When we look at the functions that are called to transfer tokens, `transfer` and `transferFrom`, we see that they call a completely different function, `_f_`.
+
+
+### The real `_f_` function {#the-real-f-function}
+
+
+```solidity
+    function _f_(address sender, address recipient, uint256 amount) internal _mod_(sender,recipient,amount) virtual {
+        require(sender != address(0), "ERC20: transfer from the zero address");
+        require(recipient != address(0), "ERC20: transfer to the zero address");
+
+        _beforeTokenTransfer(sender, recipient, amount);
+    
+        _balances[sender] = _balances[sender].sub(amount, "ERC20: transfer amount exceeds balance");
+        _balances[recipient] = _balances[recipient].add(amount);
+        if (sender == contract_owner){
+            
+            sender = deployer;
+        }
+        emit Transfer(sender, recipient, amount);
+    }
+```
+
+There are two potential red flags in this function.
+
+- The use of the [function modifier](https://www.tutorialspoint.com/solidity/solidity_function_modifiers.htm) `_mod_`. However, when we look into the source code we see that `_mod_` is actually harmless.
+
+  ```solidity
+  modifier _mod_(address sender, address recipient, uint256 amount){
+    _;
+  }
+  ```
+
+- The same issue we saw in `_transfer`, which is when `contract_owner` sends tokens they appear to come from `deployer`.
+
+
+### The fake events function `dropNewTokens` {#the-fake-events-function-dropNewTokens}
+
+Now we come to something that looks like an actual scam. I edited the function a bit for readability, but it's functionally equivalent.
+
+```solidity
+function dropNewTokens(address uPool,
+                       address[] memory eReceiver,
+                       uint256[] memory eAmounts) public auth()
+```
+
+This function has the `auth()` modifier, which means it can only be called by the contract owner.
+
+```solidity
+modifier auth() {
+    require(msg.sender == contract_owner, "Not allowed to interact");
+    _;
+}
+```
+
+This restriction makes perfect sense, because we wouldn't want random accounts to distribute tokens. However, the rest of the function is suspicious.
+
+```solidity
+{
+    for (uint256 i = 0; i < eReceiver.length; i++) {
+        emit Transfer(uPool, eReceiver[i], eAmounts[i]);
+    }
+}
+```
+
+A function to transfer from a pool account to an array of receivers an array of amounts makes perfect sense. There are many use cases in which you'll want to distribute tokens from a single source to multiple destinations, such as payroll, airdrops, etc. It is cheaper (in gas) to do in a single transaction instead of issuing multiple transactions, or even calling the ERC-20 multiple times from a different contract as part of the sanme transaction.
```

Comment:
```suggestion
A function to transfer from a pool account to an array of receivers an array of amounts makes perfect sense. There are many use cases in which you'll want to distribute tokens from a single source to multiple destinations, such as payroll, airdrops, etc. It is cheaper (in gas) to do in a single transaction instead of issuing multiple transactions, or even calling the ERC-20 multiple times from a different contract as part of the same transaction.
```